### PR TITLE
QT UI: Wifi chooser

### PR DESCRIPTION
**Pull request by pd0wm**

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Needed to squash the commits to rebase.

File: selfdrive/ui/qt/offroad/wifi.cc [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+  setStyleSheet(R"(
+    QLabel { font-size: 40px }
+    QPushButton { font-size: 40px }
+    * {
+      background-color: #114265;
+    }
+  )");
+
+  // TODO: Handle NetworkManager not running
```
Comment (pd0wm) [inline selfdrive/ui/qt/offroad/wifi.cc:43]: Check what of this is still todo

File: selfdrive/ui/qt/offroad/wifiManager.cc [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    }
+  }
+  args.endArray();
+
+  // Sort by strength
+  std::sort(r.begin(), r.end(), compare_by_strength);
+  return r;
+}
+
+int WifiManager::getSecurityType(QString path){
```
Comment (pd0wm) [inline selfdrive/ui/qt/offroad/wifiManager.cc:89]: Create enum for security types. For now just None, WPA, Unsupported.

File: selfdrive/ui/qt/offroad/wifiManager.cc [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+  int sflag = get_property(path, "Flags").toInt();
+  int wpaflag = get_property(path, "WpaFlags").toInt();
+  // int rsnflag = get_property(path, "RsnFlags").toInt();
+
+  if(sflag == 0){
+    return 0;
+  }else if(sflag == 1 && wpaflag < 400){
```
Comment (pd0wm) [inline selfdrive/ui/qt/offroad/wifiManager.cc:96]: Check actual wpa flag. Just take `(wpaflag | rsnflag) & <WPA bitmask>`

File: selfdrive/ui/qt/offroad/wifiManager.cc [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+  qDebug() << "Connecting to"<< n.ssid << "with username, password =" << username << "," <<password;
+  if(n.security_type==0){
+    connect_to_open(n.ssid);
+  }else if(n.security_type == 1){
+    connect_to_WPA(n.ssid, password);
+  }else{
+    qDebug() << "Network cannot be connected to; unknown security type";
+  }
+}
+void WifiManager::connect_to_open(QByteArray ssid){
```
Comment (pd0wm) [inline selfdrive/ui/qt/offroad/wifiManager.cc:128]: Combine with connect to WPA. Take security type as argument.

**Pull request merged**